### DISCUSSION TITLE
Setting a proxy IP in SingleFile

### DISCUSSION THREAD

**Discussion by rohit103** [gildas-lormeau/SingleFile#1072] [Q&A] [answered]
Hello, is there a way to setup a proxy IP in SingleFile? I will greatly appreciate any pointers on this. Thanks!

**Comment by gildas-lormeau** [comment]
You mean with SingleFile CLI?

**Reply by mayursarojdev** [reply]
Yes, in our use case we want to open the single file puppeteer instance with proxy IP that needs credentials for authentication.

**Reply by gildas-lormeau** [reply]
If you're using Chrome under the hood, you should be able to use `--browser-args` to pass the proxy address. Here's an example:

```sh
./single-file --browser-args=["--proxy-server=http://217.196.238.154"] https://www.example.com
```

FYI, I found the public proxy here: https://scrapingant.com/free-proxies/

**Reply by mayursarojdev** [reply]
Thanks for the response, this would work for proxies that do not need credentials. For the proxies that need authentication, in our vanilla puppeteer instance we could do page.authenticate({...}), so was wondering if that was possible in singleFile to pass proxy credentials also.

**Reply by gildas-lormeau** [reply]
Thank you for the info. I confirm the current code does not offer a way to pass data to `page.authenticate`. Meanwhile, maybe this technique would work? https://blog.apify.com/4-ways-to-authenticate-a-proxy-in-puppeteer-with-headless-chrome-in-2022/#4-setting-the-proxy-authorization-header

**Comment by gildas-lormeau** [comment, marked as answer]
The last version I just published on NPM (v1.0.27) includes these new options:
 - `--http-proxy-server` (e.g. "http://123.231.123.231")
 - `--http-proxy-username`  (e.g. "root")
 - `--http-proxy-password`  (e.g. "p4$$w0rd")

**Reply by mayursarojdev** [reply]
thank you for adding the option to pass proxy, will give it a test and let you know about the results.

**Reply by mayursarojdev** [reply]
Upgraded single file cli to latest version and tried using the new proxy options and it seems to work. Thank you
